From: Sean Whitton <spwhitton@spwhitton.name>
Date: Fri, 26 Feb 2016 21:45:21 -0700
Subject: patch-C++-tests

The (unpatched) C and C++ tests assume a Unicode locale, which we
cannot assume available in our build environment.  So patch them to
expect the C locale and set this locale when running the test suite in
debian/rules.
---
 test/flycheck-test.el | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/flycheck-test.el b/test/flycheck-test.el
index 0db6c1b..970afdc 100644
--- a/test/flycheck-test.el
+++ b/test/flycheck-test.el
@@ -3623,7 +3623,7 @@ of the file will be interrupted because there are too many #ifdef configurations
   (let ((flycheck-disabled-checkers '(c/c++-clang)))
     (flycheck-ert-should-syntax-check
      "language/c_c++/error.cpp" 'c++-mode
-     '(2 18 error "‘struct A’ has no member named ‘bar’"
+     '(2 18 error "'struct A' has no member named 'bar'"
          :checker c/c++-gcc))))
 
 (flycheck-ert-def-checker-test c/c++-gcc (c c++) fatal-error
@@ -3637,7 +3637,7 @@ of the file will be interrupted because there are too many #ifdef configurations
   (let ((flycheck-disabled-checkers '(c/c++-clang c/c++-cppcheck)))
     (flycheck-ert-should-syntax-check
      "language/c_c++/warning.c" 'c-mode
-     '(5 10 warning "unused variable ‘unused’" :checker c/c++-gcc)
+     '(5 10 warning "unused variable 'unused'" :checker c/c++-gcc)
      '(7 15 warning "comparison between signed and unsigned integer expressions"
          :checker c/c++-gcc)
      '(8 7 warning "#warning" :checker c/c++-gcc))))
